pkg/router/template/util/haproxy/map_entry.go

```diff
@@ -58,10 +58,16 @@ func generateEdgeReencryptMapEntry(cfg *BackendConfig) *HAProxyMapEntry {
 
 // generateHttpRedirectMapEntry generates a map entry for redirecting insecure/http hosts.
 func generateHttpRedirectMapEntry(cfg *BackendConfig) *HAProxyMapEntry {
-	if len(cfg.Host) > 0 && cfg.InsecurePolicy == routev1.InsecureEdgeTerminationPolicyRedirect {
-		return &HAProxyMapEntry{
-			Key:   templateutil.GenerateRouteRegexp(cfg.Host, cfg.Path, cfg.IsWildcard),
-			Value: cfg.Name,
+	if len(cfg.Host) > 0 {
+		if cfg.InsecurePolicy == routev1.InsecureEdgeTerminationPolicyRedirect {
+			return &HAProxyMapEntry{
+				Key:   templateutil.GenerateRouteRegexp(cfg.Host, cfg.Path, cfg.IsWildcard),
+				Value: "1",
+			}
+		} else {
+			return &HAProxyMapEntry{
+				Key:   templateutil.GenerateRouteRegexp(cfg.Host, cfg.Path, cfg.IsWildcard),
+				Value: "0"}
```

Comment:
Please fix the formatting issue with the `}` at the end of the line, consider using `switch`, and reduce code duplication.